internal/gengapic/client_init.go

```diff
@@ -135,155 +135,236 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
 	return nil
 }
 
-func (g *generator) clientInit(serv *descriptor.ServiceDescriptorProto, servName string) error {
+func (g *generator) abstractClientIntfInit(serv *descriptor.ServiceDescriptorProto, servName string) error {
 	p := g.printf
-
-	var hasLRO bool
+	p("// %sAbstractClient is an interface that defines the methods availaible from %s.", servName, g.apiName)
+	p("// It is agnostic as to the underlying transport, i.e. json+http, gRPC, or other.")
+	p("type %sAbstractClient interface {", servName)
 	for _, m := range serv.Method {
-		if g.isLRO(m) {
-			hasLRO = true
-			break
+		inType := g.descInfo.Type[*m.InputType]
+		inSpec, err := g.descInfo.ImportSpec(inType)
+		if err != nil {
+			return err
+		}
+		outType := g.descInfo.Type[*m.OutputType]
+		outSpec, err := g.descInfo.ImportSpec(outType)
+		if err != nil {
+			return err
 		}
-	}
 
-	imp, err := g.descInfo.ImportSpec(serv)
-	if err != nil {
-		return err
+		if m.GetOutputType() == emptyType {
+			p("%s(context.Context, *%s.%s, ...gax.CallOption) error",
+				m.GetName(),
+				inSpec.Name,
+				inType.GetName())
+			continue
+		}
+
+		if pf, err := g.pagingField(m); err != nil {
+			return err
+		} else if pf != nil {
+			iter, err := g.iterTypeOf(pf)
+			if err != nil {
+				return err
+			}
+			p("%s(context.Context, *%s.%s, ...gax.CallOption) *%s",
```

Comment:
Since the idea is for apps developers to code against the interface, maybe it's worth generating a brief comment for each method in the generated interface? (matching the method comments in the concrete clients?). WDYT? (We can defer this to a later PR)

Reply:
I'm definitely in favor of that. To the extent that the code is readable it will be used as de facto documentation. As responsible engineers we should lean into that.